modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/healthcheck/ping/PingHealthCheckConsensus.scala

```diff
@@ -1,49 +1,139 @@
 package org.tessellation.sdk.infrastructure.healthcheck.ping
 
 import cats.effect._
+import cats.effect.std.Random
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
 import cats.syntax.functor._
+import cats.syntax.traverse._
+import cats.syntax.traverseFilter._
+
+import scala.concurrent.duration._
 
 import org.tessellation.effects.GenUUID
+import org.tessellation.schema.node.NodeState
+import org.tessellation.schema.peer.Peer.toP2PContext
 import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.config.types.HealthCheckConfig
 import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
 import org.tessellation.sdk.domain.gossip.Gossip
 import org.tessellation.sdk.domain.healthcheck.consensus.types.ConsensusRounds
 import org.tessellation.sdk.domain.healthcheck.consensus.{HealthCheckConsensus, HealthCheckConsensusDriver}
+import org.tessellation.sdk.http.p2p.clients.NodeClient
+import org.tessellation.sdk.infrastructure.healthcheck.ping.{PeerAvailable, PeerUnavailable}
+
+import eu.timepit.refined.auto.autoUnwrap
+import eu.timepit.refined.types.numeric.PosInt
 
-class PingHealthCheckConsensus[F[_]: Async: GenUUID: Ref.Make](
+class PingHealthCheckConsensus[F[_]: Async: GenUUID: Random](
   clusterStorage: ClusterStorage[F],
   selfId: PeerId,
   driver: HealthCheckConsensusDriver[PingHealthCheckKey, PingHealthCheckStatus, PingConsensusHealthStatus],
+  config: HealthCheckConfig,
```

Comment:
Maybe passing `pingConfig` is enough here?

Reply:
No, because I need to pass full config to HealthCheckConsensus (see extend)